File: C_plus_plus/Palindrome_Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) return false;
        int temp = x,lenth = 1,digits = 1;
        while(temp/10){
            temp /= 10;
            ++lenth;
            digits *= 10;
        }
        for(int i = 0; i< lenth/2 ; ++i){
            if(x%10!=(temp=x/digits)) return false;
            x /= 10;
            x -= temp*digits/10;
            digits /=100;
        }
        return true;
    }
};

File: java/Palindrome_Number.java
// compare the reverse value with the origin value
public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 ) return false ;
        int revers = 0 , t = x;
        while ( t > 0 ){
            revers = revers * 10 + t % 10 ;
            t /= 10 ;
        }
        if ( x == revers) return true ;
        return false ;
    }
}
